cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,567 @@
+    """
+    Single-qubit state tomography.
+
+    The density matrix of the output state of a circuit is measured by first
+    doing projective measurements in the z-basis, which determine the
+    diagonal elements of the matrix. A X/2 or Y/2 rotation is then added before
+    the z-basis measurement, which determines the imaginary and real parts of
+    the off-diagonal matrix elements, respectively.
+
+    See Vandersypen and Chuang, Rev. Mod. Phys. 76, 1037 for details.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubit: The qubit under test.
+        circuit: The circuit to execute on the qubit before tomography.
+        num_shots: The number of measurements for each basis rotation.
+
+    Returns:
+        A StateTomographyResult object that stores and plots the density matrix.
+    """
+    circuit_z = circuit.copy()
+    circuit_z.append(ops.measure(qubit, key='z'))
+    results = sampler.run(circuit_z, repetitions=num_shots)
+    rho_11 = numpy.mean(results.measurements['z'])
+    rho_00 = 1.0 - rho_11
+
+    circuit_x = circuit.copy()
+    circuit_x.append(ops.X(qubit) ** 0.5)
+    circuit_x.append(ops.measure(qubit, key='z'))
+    results = sampler.run(circuit_x, repetitions=num_shots)
+    rho_01_im = numpy.mean(results.measurements['z']) - 0.5
+
+    circuit_y = circuit.copy()
+    circuit_y.append(ops.Y(qubit) ** -0.5)
+    circuit_y.append(ops.measure(qubit, key='z'))
+    results = sampler.run(circuit_y, repetitions=num_shots)
+    rho_01_re = 0.5 - numpy.mean(results.measurements['z'])
+
+    rho_01 = rho_01_re + 1j * rho_01_im
+    rho_10 = numpy.conj(rho_01)
+
+    rho = numpy.array([[rho_00, rho_01], [rho_10, rho_11]])
+
+    return StateTomographyResult(rho)
+
+
+def two_qubit_state_tomography(sampler: sim.SimulatesSamples,
+                               q_0: devices.GridQubit, q_1: devices.GridQubit,
+                               circuit: circuits.Circuit, num_shots: int) \
+        -> StateTomographyResult:
+    """
+    Two-qubit state tomography.
+
+    The density matrix of the output state of a two-qubit circuit is
+    determined by z-basis measurements preceded by nine different
+    combinations of I, X/2 and Y/2 operations on the two qubits. The sixteen
+    elements of the density matrix are obtained from the least-square
+    solution to a system of linear equations that may be constructed from the
+    measurement results.
+
+    See Vandersypen and Chuang, Rev. Mod. Phys. 76, 1037 for details.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        q_0: The first qubit under test.
+        q_1: The second qubit under test.
+        circuit: The circuit to execute on the qubits before tomography.
+        num_shots: The number of measurements for each basis rotation.
+
+    Returns:
+        A StateTomographyResult object that stores and plots the density matrix.
+    """
+
+    def _measurement(circ):
```

Comment:
You are shadowing the outside variable called circ.

Reply:
Shadowing removed after renaming variables